protected/humhub/modules/comment/models/forms/CommentForm.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace humhub\modules\comment\models\forms;
+
+use humhub\modules\comment\models\Comment;
+use humhub\modules\content\components\ContentActiveRecord;
+use Yii;
+use yii\web\ServerErrorHttpException;
+
+/**
+ * CommentForm
+ * @package humhub\modules\comment\models\forms
+ *
+ * @since 0.5
+ */
+class CommentForm extends yii\base\Model
+{
+    /**
+     * Comment message
+     * @var string
+     */
+    public $message;
+
+    /**
+     * The list of files attached to a comment
+     * @var array
+     */
+    public $fileList;
+
+    /**
+     * @var Comment The model to comment
+     */
+    public $comment;
+
+    /**
+     * @var Comment|ContentActiveRecord The model to comment or other active content
+     */
+    public $target;
+
+    public function __construct($target, $comment)
+    {
+        $this->target = $target;
+        $this->comment = $comment;
+        parent::__construct();
+    }
+
+    /**
+     * @return array validation rules for model attributes.
+     */
+    public function rules()
+    {
+        return [
+            [['message'], 'required', 'isEmpty' => function ($message) {
+                $hasFile = !empty($this->fileList) || (!$this->comment->isNewRecord && $this->comment->fileManager->find()->count());
```

Comment:
We should add the following before the file check in order to prevent file query on every save:

```
if(!empty($message) {
   return;
}
```

Reply:
```
if(!empty($message)) {
   return;
}
```

This check contradicting to the issue point because we will accept the empty message and attached file. That was the point of the issue. We **have to** check wherever the file attached or not on every save. Because if the message empty and file attached we approve the validation.

Reply:
I mean adding this block in the first line of the validator function. We only need to check (and query) for the existence of files if the message is empty. If the message is not empty we know its a valid comment and can skip the file query.

Reply:
Ah, ok did not see the `isEmpty`, then you probably just have to return `false` in this case? Shouldn't this function return a boolean value anyways due to https://www.yiiframework.com/doc/api/2.0/yii-validators-validator#isEmpty()-detail or you could just implement a custom validator function instead https://www.yiiframework.com/doc/guide/2.0/en/input-validation#inline-validators

Reply:
Didn't test, but perhaps like this:

```
[['message'], 'required', 'isEmpty' => function ($message) {
    if(!empty($message)) {
        return false;
    }
   
    // Allow empty message only with file attachments
    if(!empty($this->fileList) || (!$this->comment->isNewRecord && $this->comment->fileManager->find()->count())) {
        return false;
    }

    throw new ServerErrorHttpException(Yii::t('CommentModule.base', 'The comment must not be empty!'));
}],
```

Reply:
Done!